src/SystemView.cpp

```diff
@@ -28,26 +30,106 @@ static const float WHEEL_SENSITIVITY = .1f;		// Should be a variable in user set
 // i don't know how to name it
 static const double ROUGH_SIZE_OF_TURD = 10.0;
 
-TransferPlanner::TransferPlanner() {
+TransferPlanner::TransferPlanner() :
+	m_position(0., 0., 0.), m_velocity(0., 0., 0.)
+{
 	m_dvPrograde = 0.0;
 	m_dvNormal = 0.0;
 	m_dvRadial = 0.0;
+	m_startTime = 0.0;
 	m_factor = 1;
 }
 
-vector3d TransferPlanner::GetVel() { return Pi::player->GetVelocity() + GetOffsetVel(); }
+vector3d TransferPlanner::GetVel() const { return m_velocity + GetOffsetVel(); }
+
+vector3d TransferPlanner::GetOffsetVel() const {
+	if(m_position.ExactlyEqual(vector3d(0., 0., 0.)))
+		return vector3d(0., 0., 0.);
+
+	const vector3d pNormal = m_position.Cross(m_velocity);
+
+	return m_dvPrograde * m_velocity.Normalized() +
+		   m_dvNormal   * pNormal.Normalized() +
+		   m_dvRadial   * m_position.Normalized();
+}
+
```

Comment:
Maybe indentation here is a bit strange. mix of tab and spaces. Nothing worth fixing now, but to keep in mind.

Also, I _think_ we prefer to have opening and closing curly `{...}` on the same colum. @fluffyfreak scream if I'm wrong? Difficult to know when the existing pioneer code is pretty mixed. I think we had something on our wiki on coding style, but cant find it now.